File: RepoTransBench/target_projects/Python/C++/scrapy-plugins_scrapy-jsonrpc/public_tests/test_public_jsonrpc.cpp
#include <gtest/gtest.h>
#include <nlohmann/json.hpp>
#include "src/lib/jsonrpc.h"

TEST(PublicJsonRpcTest, JsonRpcSuccessObj) {
    auto result = jsonrpc_success_obj("abcde", {{"value", 99}});
    EXPECT_EQ(result["jsonrpc"], "2.0");
    EXPECT_EQ(result["id"], "abcde");
    EXPECT_EQ(result["result"]["value"], 99);
}

TEST(PublicJsonRpcTest, JsonRpcErrorObj) {
    auto error = jsonrpc_error_obj("xyz01", -123, "Unexpected Error");
    EXPECT_EQ(error["jsonrpc"], "2.0");
    EXPECT_EQ(error["id"], "xyz01");
    auto err = error["error"];
    EXPECT_EQ(err["code"], -123);
    EXPECT_EQ(err["message"], "Unexpected Error");
}

TEST(PublicJsonRpcTest, JsonRpcErrorObjWithData) {
    auto error = jsonrpc_error_obj("ab10", -20, "Message", nlohmann::json({{"details", "extra"}}));
    EXPECT_EQ(error["jsonrpc"], "2.0");
    EXPECT_EQ(error["id"], "ab10");
    auto err = error["error"];
    EXPECT_EQ(err["code"], -20);
    EXPECT_EQ(err["message"], "Message");
    EXPECT_EQ(err["data"]["details"], "extra");
}

File: RepoTransBench/target_projects/Python/Rust/scrapy-plugins_scrapy-jsonrpc/public_tests/test_public_jsonrpc.rs
use scrapy_jsonrpc::jsonrpc::{jsonrpc_success_obj, jsonrpc_error_obj};
use serde_json::{json, Value};

#[test]
fn test_jsonrpc_success_obj_public() {
    let result = jsonrpc_success_obj("abcde", json!({"value": 99}));
    assert_eq!(result["jsonrpc"], "2.0");
    assert_eq!(result["id"], "abcde");
    assert_eq!(result["result"], json!({"value": 99}));
}

#[test]
fn test_jsonrpc_error_obj_public() {
    let error = jsonrpc_error_obj("xyz01", -123, "Unexpected Error", None);
    assert_eq!(error["jsonrpc"], "2.0");
    assert_eq!(error["id"], "xyz01");
    let err = &error["error"];
    assert_eq!(err["code"], -123);
    assert_eq!(err["message"], "Unexpected Error");
}

#[test]
fn test_jsonrpc_error_obj_with_data_public() {
    let error = jsonrpc_error_obj("ab10", -20, "Message", Some(json!({"details": "extra"})));
    assert_eq!(error["jsonrpc"], "2.0");
    assert_eq!(error["id"], "ab10");
    let err = &error["error"];
    assert_eq!(err["code"], -20);
    assert_eq!(err["message"], "Message");
    assert_eq!(err["data"], json!({"details": "extra"}));
}